cloudsync/cs.py

```diff
@@ -247,38 +243,34 @@ def start(self, *, daemon=True, **kwargs):
         """
         Starts the cloudsync service.
         """
-        # Replaces :py:meth:`cloudsync.Runnable.start` so that events are processed asynchronously.
-        self.sthread = threading.Thread(target=self.smgr.run, kwargs={'sleep': 0.1, **kwargs}, daemon=daemon)
-        self.ethreads = (
-            threading.Thread(target=self.emgrs[0].run, kwargs={'sleep': self.sleep[0], **kwargs}, daemon=daemon),
-            threading.Thread(target=self.emgrs[1].run, kwargs={'sleep': self.sleep[1], **kwargs}, daemon=daemon)
-        )
-        self.sthread.start()
-        self.ethreads[0].start()
-        self.ethreads[1].start()
-        log.debug('Starting the notification manager')
+        log.debug("starting sync: %s", self.storage_label())
         self.nmgr.notify(Notification(SourceEnum.SYNC, NotificationType.STARTED, None))
-        self.nmgr.start(**kwargs)
-
+        self.smgr.start(daemon=daemon, sleep=0.1, **kwargs)
+        self.emgrs[0].start(daemon=daemon, sleep=self.sleep[0], **kwargs)
+        self.emgrs[1].start(daemon=daemon, sleep=self.sleep[1], **kwargs)
+        self.nmgr.start(daemon=daemon, **kwargs)
 
     def stop(self, forever=True, wait=True):
         """
         Stops the cloudsync service.
 
         Args:
             forever: If false is passed, then handles are left open for a future start.  Generally used for tests only.
+            wait: If false, manager threads are signaled to stop, but are NOT joined
         """
         log.info("stopping sync: %s", self.storage_label())
-        self.smgr.stop(forever=forever, wait=wait)
-        self.emgrs[0].stop(forever=forever, wait=wait)
-        self.emgrs[1].stop(forever=forever, wait=wait)
-        self.nmgr.stop(forever=forever, wait=wait)
-        if self.sthread:
-            self.sthread.join()
-            self.ethreads[0].join()
-            self.ethreads[1].join()
-            self.nmgr.wait()  # TODO: wait() above instead of join()?
-            self.sthread = None
+        for mgr in self._runnables:
+            mgr.stop(forever=forever, wait=wait)
+
+    @staticmethod
+    def stop_all(syncs: List["CloudSync"]):
+        """
+        Convenience function for stopping multiple CloudSyncs efficiently.
+        """
+        for cs in syncs:
```

Comment:
this can be in runnable:  
```
Runnable.stop_all([list])
 for ent in list:
   ent.stop(wait=False)
 for ent in list:
   ent.wait()
```

Reply:
this is right solution, I think -- since CloudSync is-a Runnable (didn't realize that until @earthman1 pointed it out) we can just pass a list of CloudSyncs to this func in Runnable, and it will do the right thing, without any additional threads.

Reply:
done